tests/test_resolver.py

```diff
@@ -626,3 +627,26 @@ def test_duplicate_requirements_issues_1550():
     distribution = installed_distribution.distribution
     assert "PyJWT" == distribution.project_name
     assert "1.7.1" == distribution.version
+
+
+def test_check_resolve_prerelease_transitive_dependencies_issue_1730(tmpdir):
+    # type: (Any) -> None
+
+    indirect_wheel = build_wheel(name="indirect", version="2.12.0.dev3")
```

Comment:
Because of the note above about the IT not actually being exercised in CI, I added a simulation here. This does fail prior to the fix with:
```
E           Unsatisfiable: Failed to resolve compatible distributions:
E           1: direct==2.12.0.dev3 requires indirect==2.12.0.dev3 but indirect 2.12.0.dev3 was resolved
```